pynamodb/attributes.pyi

```diff
@@ -145,7 +145,6 @@ class MapAttribute(Generic[_KT, _VT], Attribute[Mapping[_KT, _VT]], metaclass=Ma
     @overload
     def __get__(self: _MT, instance: Any, owner: Any) -> _MT: ...
     def is_type_safe(self, key: Any, value: Any) -> bool: ...
-    def validate(self) -> bool: ...
```

Comment:
Are we changing API contract here?

Reply:
good point, what do you suggest here? recursive check as well?

Reply:
I removed it because the old same level check is no longer used, but can definitely add it back if it changes the behavior too much to certain users.

Reply:
I removed it because the old same level check is no longer used, but can definitely add it back if it changes the behavior too much for certain users.

Reply:
`validate` and `is_type_safe` do not start with an underscore, so they're part of the API. It's hard to say how much they're used in practice, but we should probably bump the major version if we remove them. Is it possible to separate that change from what you're doing? Is the previous way, where this method was used during serialization, not suitable anymore?